api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -56,15 +56,37 @@ type OpenstackDataPlaneServiceCert struct {
 	EDPMRoleServiceName string `json:"edpmRoleServiceName,omitempty"`
 }
 
+// OpenStackDataPlaneConfigMap defines the properties of the ConfigMaps used
+// by the service.
+type OpenStackDataPlaneConfigMap struct {
+	// Name of the ConfigMap
+	Name string `json:"name"`
+
+	// Required is whether the ConfigMap is required to exist
+	// +kubebuilder:validation:Optional
+	Required bool `json:"required,omitempty"`
+}
+
+// OpenStackDataPlaneSecret defines the properties of the Secrets used by
+// the service.
+type OpenStackDataPlaneSecret struct {
+	// Name of the Secret
+	Name string `json:"name"`
+
+	// Required is whether the Secret is required to exist
+	// +kubebuilder:validation:Optional
+	Required bool `json:"required,omitempty"`
+}
```

Comment:
As a follow up I would try to validate in the webhook that a service with EDPMServiceType=nova:
* runs the nova playbook
* has the cell secret as required
  * nova-operator generates the secret with a name that is based on the name of the cell. But knowing the cell names require looking up the OpenStackControlPlane CR or more likely the NovaCell CR which would be an unwanted dependency. 
* has the ssh migration key secret as required. This is even more problematic as the name of the secret does not matter for the edpm-ansible code. The only real requirement is to have an ssh keypair with filenames (secret keys) ssh-privatekey and ssh-publickey provided.

Maybe this is still impractical to do these kind of validations :/ I guess I need to think more how can we express the needs of the edm-nova role during the service creation / service deployment.  Anyhow this is not a real comment on the current struct proposal so feel free to ignore it.